tests/test_sns.py

```diff
@@ -611,3 +611,69 @@ def test_sns_delete(self):
         client = session_factory().client('sns')
         resources = client.list_topics()['Topics']
         self.assertEqual(len(resources), 0)
+
+    def test_sns_tag(self):
+        session_factory = self.replay_flight_data("test_sns_tag")
+        p = self.load_policy(
+            {
+                "name": "tag-sns",
+                "resource": "sns",
+                "filters": [{"tag:Tagging": "absent"}],
+                "actions": [{"type": "tag", "key": "Tagging", "value": "added"}],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        client = session_factory().client("sns")
+        tags = client.list_tags_for_resource(ResourceArn=resources[0]["TopicArn"])["Tags"]
+        self.assertEqual(tags[0]["Value"], "added")
+
+    def test_sns_remove_tag(self):
+        session_factory = self.replay_flight_data(
+            "test_sns_remove_tag")
+        p = self.load_policy(
+            {
+                "name": "untag-sns",
+                "resource": "sns",
+                "filters": [{"tag:Tagging": "added"}],
+                "actions": [{"type": "remove-tag", "tags": ["Tagging"]}],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        client = session_factory().client("sns")
+        tags = client.list_tags_for_resource(ResourceArn=resources[0]["TopicArn"])["Tags"]
+        self.assertEqual(len(tags), 0)
+
+    def test_sns_mark_for_op(self):
```

Comment:
there's two topics in the account so there's some additional data recorded, can you re-record with only 1 topic

Reply:
looks like this applies to all three of the tests you wrote here

Reply:
Thanks, pushed new recorded test with one resource